code/datums/diseases/zombie.dm

```diff
@@ -70,29 +72,30 @@ var/list/zombie_cure = list()
 				else if(prob(5))
 					affected_mob << "<span class='danger'>You feel a stabbing pain in your head.</span>"
 					affected_mob.confused += 10
-				else if(prob(5) && ishuman(affected_mob))
-					var/mob/living/carbon/human/H = affected_mob
-					H.vessel.remove_reagent("blood",rand(1,2))
+				else if(prob(5))
+					H.vessel.remove_reagent("blood",rand(5,15))
 					affected_mob.visible_message("<span class='warning'>[affected_mob] looks a bit pale...</span>", "<span class='notice'>You look a bit pale...</span>")
 				else if(prob(5))
 					affected_mob << "<span class='notice'>[pick("You feel hungry.", "You crave for food.")]</span>"
 					affected_mob.overeatduration = max(affected_mob.overeatduration - 25, 0)
 					affected_mob.nutrition = max(affected_mob.nutrition - 25, 0)
 			if(4)
+				var/mob/living/carbon/human/H = affected_mob
```

Comment:
I know it's very unlikely for affect_mob to be anything but a human, but is it possible to add a return if !H?